### PR TITLE
Update UI Frameworks - Group 6: Many Small Files

### PR DESCRIPTION
**Pull request by rayking29**
# Fixes or Implements [AB#15557](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15557)

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by BrianMaki — APPROVED** [review]

**Review by BronzBierd — DISMISSED** [review]
Didn't notice any issues.

File: Apps/WebClient/src/ClientApp/src/components/report/ImmunizationHistoryReportComponent.vue [hunk 1]
```diff
@@ -141,122 +193,69 @@ export default class ImmunizationHistoryReportComponent extends Vue {
...
+    });
+}
+
+watch(isLoading, () => {
+    emit("on-is-loading-changed", isLoading.value);
+});
+
+function onIsEmptyChanged(): void {
+    emit("on-is-empty-changed", isEmpty.value && isRecommendationEmpty.value);
 }
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/report/ImmunizationHistoryReportComponent.vue:241]: Nit: swap order to keep functions and watch() statements grouped separately.

```suggestion
function onIsEmptyChanged(): void {
    emit("on-is-empty-changed", isEmpty.value && isRecommendationEmpty.value);
}

watch(isLoading, () => {
    emit("on-is-loading-changed", isLoading.value);
});
```

File: Apps/WebClient/src/ClientApp/src/components/modal/ProtectiveWordComponent.vue [hunk 2]
```diff
@@ -10,79 +9,80 @@ import container from "@/plugins/container";
...
+}
 
-    private handleOk(bvModalEvt: Event): void {
-        // Prevent modal from closing
-        bvModalEvt.preventDefault();
-        this.fetchMedications();
-    }
+function handleHidden(): void {
+    protectiveWord.value = "";
+    isDismissed.value = false;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/modal/ProtectiveWordComponent.vue:63]:
```suggestion
    isDismissed.value = true;
```

File: Apps/WebClient/src/ClientApp/src/components/timeline/entryCard/EntrycardTimelineComponent.vue [hunk 3]
```diff
@@ -15,8 +15,8 @@ library.add(farComment, faPaperclip);
 interface Props {
     cardId: string;
     entry: TimelineEntry;
-    entryIcon: string;
-    title: string;
+    title?: string;
+    entryIcon?: string;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/timeline/entryCard/EntrycardTimelineComponent.vue:19]:
```suggestion
    title: string;
    entryIcon?: string;
```

File: Apps/WebClient/src/ClientApp/src/components/timeline/entryCard/EntrycardTimelineComponent.vue [hunk 4]
```diff
@@ -25,6 +25,8 @@ interface Props {
     subtitle?: string;
 }
 const props = withDefaults(defineProps<Props>(), {
+    title: "",
+    entryIcon: "question",
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/timeline/entryCard/EntrycardTimelineComponent.vue:29]:
```suggestion
    entryIcon: "question",
```

**Review by MikeLyttle — APPROVED** [review]

**Review by BronzBierd — APPROVED** [review]

**Pull request merged**